internal/imgmanager/image_puller.go

```diff
@@ -0,0 +1,284 @@
+package imgmanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	eventtypes "github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/v2/core/events"
+	"github.com/containerd/typeurl/v2"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+
+	ofenv1 "github.com/cybozu-go/ofen/api/v1"
+)
+
+type NodeImageSetStatus struct {
+	Images sync.Map
```

Comment:
I think some `ImagePullStatus` will leak when NodeImageSet is updated and some images in the NodeImageSet are deleted. Is that right?